unsteady/solver.py

```diff
@@ -757,23 +601,21 @@ def copy_data_from_intermediary_mesh(self, i):
             self.fwd_solutions_bathymetry[i].assign(self.intermediary_solutions_bathymetry[i])
 
         if hasattr(self.op, 'sediment_model'):
-            self.op.sediment_model.old_bathymetry_2d.assign(self.intermediary_solutions_old_bathymetry[i])
+            self.op.sediment_model.old_bathymetry_2d.project(self.intermediary_solutions_old_bathymetry[i])
             self.op.sediment_model.uv_cg.assign(self.intermediary_solutions_uv_cg[i])
-            self.op.sediment_model.bed_stress.assign(self.intermediary_solutions_bed_stress[i])
-            self.op.sediment_model.depth.assign(self.intermediary_solutions_depth[i])
+            self.op.sediment_model.bed_stress.project(self.intermediary_solutions_bed_stress[i])
+            self.op.sediment_model.depth.project(self.intermediary_solutions_depth[i])
             if self.op.suspended:
                 if self.op.convective_vel_flag:
-                    self.op.sediment_model.corr_factor_model.corr_vel_factor.assign(self.intermediary_corr_vel_factor[i])
-                self.op.sediment_model.ceq.assign(self.intermediary_ceq[i])
-                self.op.sediment_model.equiltracer.assign(self.intermediary_equiltracer[i])
+                    self.op.sediment_model.corr_factor_model.corr_vel_factor.project(self.intermediary_corr_vel_factor[i])
+                self.op.sediment_model.ceq.project(self.intermediary_ceq[i])
+                self.op.sediment_model.equiltracer.project(self.intermediary_equiltracer[i])
                 self.op.sediment_model.ero.assign(self.intermediary_ero[i])
-                self.op.sediment_model.ero_term.assign(self.intermediary_ero_term[i])
-                self.op.sediment_model.depo_term.assign(self.intermediary_depo_term[i])
             if self.op.bedload:
-                self.op.sediment_model.calfa.assign(self.intermediary_solutions_calfa[i])
-                self.op.sediment_model.salfa.assign(self.intermediary_solutions_salfa[i])
+                self.op.sediment_model.calfa.project(self.intermediary_solutions_calfa[i])
+                self.op.sediment_model.salfa.project(self.intermediary_solutions_salfa[i])
                 if self.op.angle_correction:
-                    self.op.sediment_model.stress.assign(self.intermediary_solutions_stress[i])
+                    self.op.sediment_model.stress.project(self.intermediary_solutions_stress[i])
```

Comment:
I think this is overwriting what you just did in the last PR?

Reply:
Yes - unfortunately I did some tests and it turns out for some reason the ones I changed really do have to be projects rather than assign